pkg/store/prompb/remote.proto

```diff
@@ -0,0 +1,76 @@
+// Copyright 2016 Prometheus Team
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// NOTE(fabxc): this is just hard-copied from prometheus/prometheus/prompb
+// since depending on the generated package just added dependency complexities.
+// It is also generated in a way that makes heavy use of pointers, e.g. for each sample
+// which we want to avoid for ourselves.
+
+syntax = "proto3";
+package prometheus;
+
+import "gogoproto/gogo.proto";
+
+option (gogoproto.sizer_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (gogoproto.goproto_getters_all) = false;
+
+option go_package = "prompb";
+
+message ReadRequest {
+  repeated Query queries = 1 [(gogoproto.nullable) = false];
+}
+
+message ReadResponse {
+  // In same order as the request's queries.
+  repeated QueryResult results = 1 [(gogoproto.nullable) = false];
+}
+
+message Query {
```

Comment:
Most of these are totally the same as our StoreAPI bits.. I guess it would be not safe to reuse these? (:

Reply:
Yes, they are all pretty similar. But I don't want our stuff to depend on prompb now, when both are basically still experimental. It's really just extra effort right now.

Even later, I'm not sure how much good it brings for two essentially distinct APIs.